src/oscar/apps/voucher/abstract_models.py

```diff
@@ -177,6 +177,11 @@ def save(self, *args, **kwargs):
         self.code = self.code.upper()
         super().save(*args, **kwargs)
 
+    def delete(self, *args, **kwargs):
+        """Deletes all referenced conditional offers"""
+        self.offers.all().delete()
```

Comment:
This is too aggressive and can result in data loss. Oscar's models do not prevent the same offer being used in multiple places - so indiscriminately deleting all offers could affect other vouchers or remove an offer that is used for another purpose. 

We should do something similar to [what we do for benefits/conditions](https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/offer/signals.py#L12-L31), i.e., check that :

1. Only one offer is associated with the voucher, and it is of the `voucher` type.
2. The offer name corresponds to the voucher name.

And only then delete the offer.

I also think this should be done in a `post_delete` signal so that there is no risk of integrity errors if the delete fails for another reason.

Reply:
Thanks for your feedback, I will update the code accordingly.